test/fw/ptl/utils/pbs_crayutils.py

```diff
@@ -175,3 +175,15 @@ def get_numthreads(self, nid):
                 key = line.split()
                 numthreads = int(key[1])
         return numthreads
+
+    def num_compute_vnodes(self):
+        """
+        Count the Cray compute nodes and return the value.
+        """
+        vlist = []
+        vnl = self.server.filter(
+            VNODE, {'resources_available.vntype': 'cray_compute'})
+        vlist = vnl["resources_available.vntype=cray_compute"]
+        nv = len(vlist)
+        self.assertNotEqual(nv, 0, "There are no cray_compute vnodes present.")
```

Comment:
While I can change the docstring and the function name, I think having the utility function just to count the number of vnodes is sufficient even though it would have only one single return statement because there could be tests to confirm that the number of vnodes is equal to 0. So, I am leaving the decision to assert or not up to the user of the utility function.